src/Strategy/NuGet/ProjectAssetsJson.hs

```diff
@@ -148,49 +150,54 @@ instance FromJSON ProjectAssetsJson where
           pure (FrameworkName framework, frameworkDeps)
         pure $ Map.fromList projectFrameworks
 
+newtype NugetLabel = Env DepEnvironment deriving (Eq, Ord, Show)
+
 buildGraph :: ProjectAssetsJson -> Graphing Dependency
-buildGraph project = foldr (<>) Graphing.empty graphsOfTargetFrameworks
+buildGraph project = run . withLabeling toDependency $ graphsOfTargetFrameworks
   where
     graphsOfTargetFrameworks =
-      map
+      mapM
         (graphOfFramework $ projectFramework project)
         (Map.toList $ targets project)
 
-graphOfFramework :: Map.Map FrameworkName (Set Text) -> (FrameworkName, Map.Map NuGetDepKey DependencyInfo) -> Graphing Dependency
-graphOfFramework projectFrameworkDeps (targetFramework, targetFrameworkDeps) =
-  Graphing.gmap toDependency
-    . withoutProjects
-    . promoteToDirect isDirectDep
-    $ unfoldDeep allResolvedDeps getTransitiveDeps id
+    toDependency :: NuGetDep -> Set NugetLabel -> Dependency
+    toDependency NuGetDep{..} =
+      foldr (\_ d -> d) $
+        Dependency
+          { dependencyType = NuGetType
+          , dependencyName = depName
+          , dependencyVersion = Just (CEq depVersion)
+          , dependencyLocations = []
+          , dependencyEnvironments = mempty
+          , dependencyTags = Map.empty
+          }
+
+graphOfFramework :: Has (LabeledGrapher NuGetDep NugetLabel) sig m => (Map.Map FrameworkName (Set Text)) -> (FrameworkName, Map.Map NuGetDepKey DependencyInfo) -> m ()
```

Comment:
We aren't using labels at all here.  Can we just use the grapher effect, and then run `Graphing.gmap` to convert to dependency later?